source/client-side-encryption/tests/types.template.yml

```diff
@@ -0,0 +1,527 @@
+            cursor: {}
+            filter:
+              name: "datakeys"
+            $db: admin
+          command_name: listCollections
+      # Then key is fetched from the key vault.
+      - command_started_event:
+          command:
+            find: datakeys
+            filter: {"$or": [{"_id": {"$in": [ {{key()["_id"]}} ] }}, {"keyAltNames": {"$in": []}}]}
+            $db: admin
+          command_name: find
+      - command_started_event:
+          command:
+            insert: *collection_name
+            documents:
+              - &doc1_encrypted { _id: 1, encrypted_symbol: {{ciphertext("test", field="encrypted_symbol")}} }
+            ordered: true
+          command_name: insert
+      - command_started_event:
+          command:
+            find: *collection_name
+            filter: {_id: 1}
+          command_name: find
+    outcome:
+      collection:
+        # Outcome is checked using a separate MongoClient without auto encryption.
+        data:
+          - *doc1_encrypted
+  - description: "type=int"
+    clientOptions:
+      autoEncryptOpts:
+        schemaMap:
+          "default.default": {{schema_w_type("int")}}
+        kmsProviders:
+          aws: {} # Credentials filled in from environment.
+    operations:
+      - name: insertOne
+        arguments:
+          document: &doc2 { _id: 1, encrypted_int: {"$numberInt": "123"} }
+      - name: findOne
+        arguments:
+          filter: { _id: 1 }
+          result: *doc2
+    expectations:
+      - command_started_event:
+          command:
+            listCollections: 1
+            cursor: {}
+            filter:
+              name: "datakeys"
+            $db: admin
+          command_name: listCollections
+      # Then key is fetched from the key vault.
+      - command_started_event:
+          command:
+            find: datakeys
+            filter: {"$or": [{"_id": {"$in": [ {{key()["_id"]}} ] }}, {"keyAltNames": {"$in": []}}]}
+            $db: admin
+          command_name: find
+      - command_started_event:
+          command:
+            insert: *collection_name
+            documents:
+              - &doc2_encrypted { _id: 1, encrypted_int: {{ciphertext("test", field="encrypted_int32")}} }
+            ordered: true
+          command_name: insert
+      - command_started_event:
+          command:
+            find: *collection_name
+            filter: {_id: 1}
+          command_name: find
+    outcome:
+      collection:
+        # Outcome is checked using a separate MongoClient without auto encryption.
+        data:
+          - *doc2_encrypted
+  - description: "type=double"
+    clientOptions:
+      autoEncryptOpts:
+        schemaMap:
+          "default.default": {{schema_w_type("double")}}
+        kmsProviders:
+          aws: {} # Credentials filled in from environment.
+    operations:
+      - name: insertOne
+        arguments:
+          document: &doc4 { _id: 1, encrypted_long: {"$double": "1.23"} }
```

Comment:
This should be just:

    encrypted_long: 1.23

or if it has to be canonical:

    encrypted_long: {"$numberDouble": "1.23"}

Reply:
CC @mbroadst

Reply:
Fixed.